index.js

```diff
@@ -25,8 +25,10 @@ function singletonify (inst) {
   Object.keys(inst).forEach((key) => {
     if (key === 'argv') {
       Argv.__defineGetter__(key, inst.__lookupGetter__(key))
+    } else if (typeof inst[key] === 'function') {
+      Argv[key] = inst[key].bind(inst)
     } else {
```

Comment:
why don't we make this slightly more explicit:

`if (['$0', 'parsed'].indexOf(key) !== -1) {`

and we should be mindful about leaking any more stateful variables on the yargs object going forward.

Reply:
I don't get the point. If we support singleton mode, shouldn't every property/method call on Argv  have the same effect as if directly called on the singleton?

Reply:
fair 👍 I just noticed that, I think, `argv`, `$0`, and `parsed` are the only properties that leak ... and honestly, I don't think we meant to leak `parsed` as part of the API.

Reply:
Oops! I did not realize this... However, `parsed` is already leaking from the API when not using the singleton. So, does it have a sense to filter properties we accept to leak from the singleton, and not to do it when not using the singleton?